teku/src/main/java/tech/pegasys/teku/cli/subcommand/admin/WeakSubjectivityCommand.java

```diff
@@ -0,0 +1,120 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.cli.subcommand.admin;
+
+import static tech.pegasys.teku.infrastructure.logging.SubCommandLogger.SUB_COMMAND_LOG;
+
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+import picocli.CommandLine;
+import tech.pegasys.teku.cli.converter.PicoCliVersionProvider;
+import tech.pegasys.teku.cli.options.DataOptions;
+import tech.pegasys.teku.cli.options.DataStorageOptions;
+import tech.pegasys.teku.cli.options.NetworkOptions;
+import tech.pegasys.teku.storage.events.WeakSubjectivityState;
+import tech.pegasys.teku.storage.events.WeakSubjectivityUpdate;
+import tech.pegasys.teku.storage.server.Database;
+import tech.pegasys.teku.storage.server.VersionedDatabaseFactory;
+import tech.pegasys.teku.util.config.NetworkDefinition;
+
+@CommandLine.Command(
+    name = "weak-subjectivity",
+    description = "Commands related to weak subjectivity configuration",
+    // TODO(#2779) - Enable this command
+    hidden = true,
+    showDefaultValues = true,
+    abbreviateSynopsis = true,
+    mixinStandardHelpOptions = true,
+    versionProvider = PicoCliVersionProvider.class,
+    synopsisHeading = "%n",
+    descriptionHeading = "%nDescription:%n%n",
+    optionListHeading = "%nOptions:%n",
+    footerHeading = "%n",
+    footer = "Teku is licensed under the Apache License 2.0")
+public class WeakSubjectivityCommand implements Runnable {
+
+  @Override
+  public void run() {
+    CommandLine.usage(this, System.out);
+  }
+
+  @CommandLine.Command(
+      name = "clear-state",
+      description = "Clears stored weak subjectivity configuration",
+      mixinStandardHelpOptions = true,
+      showDefaultValues = true,
+      abbreviateSynopsis = true,
+      versionProvider = PicoCliVersionProvider.class,
+      synopsisHeading = "%n",
+      descriptionHeading = "%nDescription:%n%n",
+      optionListHeading = "%nOptions:%n",
+      footerHeading = "%n",
+      footer = "Teku is licensed under the Apache License 2.0")
+  public int clearWeakSubjectivityState(
+      @CommandLine.Mixin final DataOptions dataOptions,
+      @CommandLine.Mixin final DataStorageOptions dataStorageOptions,
+      @CommandLine.Mixin final NetworkOptions networkOptions) {
+    final Database db = createDatabase(dataOptions, dataStorageOptions, networkOptions);
+    // Pull value before updating
+    final WeakSubjectivityState original = db.getWeakSubjectivityState();
+    if (original.isEmpty()) {
+      SUB_COMMAND_LOG.display("Weak subjectivity state is already empty - nothing to clear.");
+      return 0;
```

Comment:
Should we not return 1, in this case, showing that there was an error?

Reply:
Hmm - good question.  I wasn't considering this an error ...

Reply:
I think 0 makes sense - say you have some script that clears this out, you wouldn't want it to error out just because the state was already empty.

Reply:
That's fair.